// @nodegui/react-nodegui/dist/components/Calendar/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calendar = void 0;
const config_1 = require("../config");
const RNCalendar_1 = require("./RNCalendar");
class CalendarConfig extends config_1.ComponentConfig {
    constructor() {
        super(...arguments);
        this.tagName = RNCalendar_1.RNCalendar.tagName;
    }
    shouldSetTextContent(nextProps) {
        return false;
    }
    createInstance(newProps, rootInstance, context, workInProgress) {
        const widget = new RNCalendar_1.RNCalendar();
        widget.setProps(newProps, {});
        return widget;
    }
    commitMount(instance, newProps, internalInstanceHandle) {
        if (newProps.visible !== false) {
            instance.show();
        }
        return;
    }
    commitUpdate(instance, updatePayload, oldProps, newProps, finishedWork) {
        instance.setProps(newProps, oldProps);
    }
}
/**
 * The Calendar component provides ability to add calendar widgets. It is based on
 * [NodeGui's QCalendarWidget](https://docs.nodegui.org/docs/api/generated/classes/qcalendarwidget/).
 * @example
 * ```typescriptreact
 * import {
 *   DateFormat,
 *   DayOfWeek,
 *   NativeElement,
 *   QCalendarWidgetSignals,
 *   QDate,
 * } from "@nodegui/nodegui";
 * import {
 *   Calendar,
 *   Renderer,
 *   useEventHandler,
 *   Window,
 * } from "@nodegui/react-nodegui";
 * import React from "react";
 * const App = () => {
 *   const calendarClickHandler = useEventHandler<QCalendarWidgetSignals>(
 *     {
 *       activated: (nativeDate) => {
 *         const date = new QDate(nativeDate as unknown as NativeElement);
 *         console.log(`activated: ${date.toString(DateFormat.SystemLocaleDate)}`);
 *       },
 *       clicked: (nativeDate) => {
 *         const date = new QDate(nativeDate as unknown as NativeElement);
 *         console.log(`clicked: ${date.toString(DateFormat.SystemLocaleDate)}`);
 *       },
 *       currentPageChanged: (year, month) => {
 *         console.log(`currentPageChanged: year, month: ${year}, ${month}`);
 *       },
 *       selectionChanged: () => {
 *         console.log("selectionChanged");
 *       },
 *     },
 *     []
 *   );
 *
 *   return (
 *     <Window>
 *       <Calendar firstDayOfWeek={DayOfWeek.Sunday} on={calendarClickHandler} />
 *     </Window>
 *   );
 * }
 * Renderer.render(<App />);
 * ```
 */
exports.Calendar = (0, config_1.registerComponent)(new CalendarConfig());
