// @nodegui/react-nodegui/dist/components/Calendar/RNCalendar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RNCalendar = void 0;
const nodegui_1 = require("@nodegui/nodegui");
const helpers_1 = require("../../utils/helpers");
const RNView_1 = require("../View/RNView");
const setCalendarProps = (widget, newProps, oldProps) => {
    const setter = {
        set dateEditAcceptDelay(delay) {
            widget.setDateEditAcceptDelay(delay);
        },
        set dateEditEnabled(enable) {
            widget.setDateEditEnabled(enable);
        },
        set firstDayOfWeek(dayOfWeek) {
            widget.setFirstDayOfWeek(dayOfWeek);
        },
        set font(font) {
            widget.setFont(font);
        },
        set gridVisible(show) {
            widget.setGridVisible(show);
        },
        set horizontalHeaderFormat(format) {
            widget.setHorizontalHeaderFormat(format);
        },
        set navigationBarVisible(visible) {
            widget.setNavigationBarVisible(visible);
        },
        set selectionMode(mode) {
            widget.setSelectionMode(mode);
        },
        set verticalHeaderFormat(format) {
            widget.setVerticalHeaderFormat(format);
        },
    };
    Object.assign(setter, newProps);
    (0, RNView_1.setViewProps)(widget, newProps, oldProps);
};
/**
 * @ignore
 */
class RNCalendar extends nodegui_1.QCalendarWidget {
    setProps(newProps, oldProps) {
        setCalendarProps(this, newProps, oldProps);
    }
    appendInitialChild(child) {
        (0, helpers_1.throwUnsupported)(this);
    }
    appendChild(child) {
        (0, helpers_1.throwUnsupported)(this);
    }
    insertBefore(child, beforeChild) {
        (0, helpers_1.throwUnsupported)(this);
    }
    removeChild(child) {
        (0, helpers_1.throwUnsupported)(this);
    }
}
exports.RNCalendar = RNCalendar;
RNCalendar.tagName = "calendar";
