// @nodegui/react-nodegui/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.appProxy = exports.hot = exports.Renderer = exports.useEventHandler = exports.ErrorPrompt = exports.ListItem = exports.List = exports.TableItem = exports.Table = exports.ProgressDialog = exports.InputDialog = exports.FontDialog = exports.FileDialog = exports.ColorDialog = exports.Dialog = exports.TabItem = exports.Tabs = exports.SystemTrayIcon = exports.ComboBox = exports.ScrollArea = exports.SpinBox = exports.Dial = exports.RadioButton = exports.ProgressBar = exports.Calendar = exports.PlainTextEdit = exports.MenuBar = exports.Menu = exports.LineEdit = exports.CheckBox = exports.Button = exports.AnimatedImage = exports.Image = exports.Text = exports.Window = exports.View = exports.Slider = exports.GridColumn = exports.GridRow = exports.GridView = exports.BoxView = exports.Action = void 0;
var Action_1 = require("./components/Action");
Object.defineProperty(exports, "Action", { enumerable: true, get: function () { return Action_1.Action; } });
var BoxView_1 = require("./components/BoxView");
Object.defineProperty(exports, "BoxView", { enumerable: true, get: function () { return BoxView_1.BoxView; } });
var GridView_1 = require("./components/GridView");
Object.defineProperty(exports, "GridView", { enumerable: true, get: function () { return GridView_1.GridView; } });
var GridRow_1 = require("./components/GridView/GridRow");
Object.defineProperty(exports, "GridRow", { enumerable: true, get: function () { return GridRow_1.GridRow; } });
var GridColumn_1 = require("./components/GridView/GridColumn");
Object.defineProperty(exports, "GridColumn", { enumerable: true, get: function () { return GridColumn_1.GridColumn; } });
var Slider_1 = require("./components/Slider");
Object.defineProperty(exports, "Slider", { enumerable: true, get: function () { return Slider_1.Slider; } });
var View_1 = require("./components/View");
Object.defineProperty(exports, "View", { enumerable: true, get: function () { return View_1.View; } });
var Window_1 = require("./components/Window");
Object.defineProperty(exports, "Window", { enumerable: true, get: function () { return Window_1.Window; } });
var Text_1 = require("./components/Text");
Object.defineProperty(exports, "Text", { enumerable: true, get: function () { return Text_1.Text; } });
var Image_1 = require("./components/Image");
Object.defineProperty(exports, "Image", { enumerable: true, get: function () { return Image_1.Image; } });
var AnimatedImage_1 = require("./components/AnimatedImage");
Object.defineProperty(exports, "AnimatedImage", { enumerable: true, get: function () { return AnimatedImage_1.AnimatedImage; } });
var Button_1 = require("./components/Button");
Object.defineProperty(exports, "Button", { enumerable: true, get: function () { return Button_1.Button; } });
var CheckBox_1 = require("./components/CheckBox");
Object.defineProperty(exports, "CheckBox", { enumerable: true, get: function () { return CheckBox_1.CheckBox; } });
var LineEdit_1 = require("./components/LineEdit");
Object.defineProperty(exports, "LineEdit", { enumerable: true, get: function () { return LineEdit_1.LineEdit; } });
var Menu_1 = require("./components/Menu");
Object.defineProperty(exports, "Menu", { enumerable: true, get: function () { return Menu_1.Menu; } });
var MenuBar_1 = require("./components/MenuBar");
Object.defineProperty(exports, "MenuBar", { enumerable: true, get: function () { return MenuBar_1.MenuBar; } });
var PlainTextEdit_1 = require("./components/PlainTextEdit");
Object.defineProperty(exports, "PlainTextEdit", { enumerable: true, get: function () { return PlainTextEdit_1.PlainTextEdit; } });
var Calendar_1 = require("./components/Calendar");
Object.defineProperty(exports, "Calendar", { enumerable: true, get: function () { return Calendar_1.Calendar; } });
var ProgressBar_1 = require("./components/ProgressBar");
Object.defineProperty(exports, "ProgressBar", { enumerable: true, get: function () { return ProgressBar_1.ProgressBar; } });
var RadioButton_1 = require("./components/RadioButton");
Object.defineProperty(exports, "RadioButton", { enumerable: true, get: function () { return RadioButton_1.RadioButton; } });
var Dial_1 = require("./components/Dial");
Object.defineProperty(exports, "Dial", { enumerable: true, get: function () { return Dial_1.Dial; } });
var SpinBox_1 = require("./components/SpinBox");
Object.defineProperty(exports, "SpinBox", { enumerable: true, get: function () { return SpinBox_1.SpinBox; } });
var ScrollArea_1 = require("./components/ScrollArea");
Object.defineProperty(exports, "ScrollArea", { enumerable: true, get: function () { return ScrollArea_1.ScrollArea; } });
var ComboBox_1 = require("./components/ComboBox");
Object.defineProperty(exports, "ComboBox", { enumerable: true, get: function () { return ComboBox_1.ComboBox; } });
var SystemTrayIcon_1 = require("./components/SystemTrayIcon");
Object.defineProperty(exports, "SystemTrayIcon", { enumerable: true, get: function () { return SystemTrayIcon_1.SystemTrayIcon; } });
var Tab_1 = require("./components/Tab");
Object.defineProperty(exports, "Tabs", { enumerable: true, get: function () { return Tab_1.Tabs; } });
var TabItem_1 = require("./components/TabItem");
Object.defineProperty(exports, "TabItem", { enumerable: true, get: function () { return TabItem_1.TabItem; } });
var Dialog_1 = require("./components/Dialog");
Object.defineProperty(exports, "Dialog", { enumerable: true, get: function () { return Dialog_1.Dialog; } });
var ColorDialog_1 = require("./components/ColorDialog");
Object.defineProperty(exports, "ColorDialog", { enumerable: true, get: function () { return ColorDialog_1.ColorDialog; } });
var FileDialog_1 = require("./components/FileDialog");
Object.defineProperty(exports, "FileDialog", { enumerable: true, get: function () { return FileDialog_1.FileDialog; } });
var FontDialog_1 = require("./components/FontDialog");
Object.defineProperty(exports, "FontDialog", { enumerable: true, get: function () { return FontDialog_1.FontDialog; } });
var InputDialog_1 = require("./components/InputDialog");
Object.defineProperty(exports, "InputDialog", { enumerable: true, get: function () { return InputDialog_1.InputDialog; } });
var ProgressDialog_1 = require("./components/ProgressDialog");
Object.defineProperty(exports, "ProgressDialog", { enumerable: true, get: function () { return ProgressDialog_1.ProgressDialog; } });
var Table_1 = require("./components/Table");
Object.defineProperty(exports, "Table", { enumerable: true, get: function () { return Table_1.Table; } });
var TableItem_1 = require("./components/TableItem");
Object.defineProperty(exports, "TableItem", { enumerable: true, get: function () { return TableItem_1.TableItem; } });
var List_1 = require("./components/List");
Object.defineProperty(exports, "List", { enumerable: true, get: function () { return List_1.List; } });
var ListItem_1 = require("./components/ListItem");
Object.defineProperty(exports, "ListItem", { enumerable: true, get: function () { return ListItem_1.ListItem; } });
var ErrorPrompt_1 = require("./components/ErrorPrompt");
Object.defineProperty(exports, "ErrorPrompt", { enumerable: true, get: function () { return ErrorPrompt_1.ErrorPrompt; } });
var hooks_1 = require("./hooks");
Object.defineProperty(exports, "useEventHandler", { enumerable: true, get: function () { return hooks_1.useEventHandler; } });
var renderer_1 = require("./renderer");
Object.defineProperty(exports, "Renderer", { enumerable: true, get: function () { return renderer_1.Renderer; } });
var hot_reload_1 = require("./development/hot-reload");
Object.defineProperty(exports, "hot", { enumerable: true, get: function () { return hot_reload_1.hot; } });
Object.defineProperty(exports, "appProxy", { enumerable: true, get: function () { return hot_reload_1.appProxy; } });
